### PR TITLE
Issue 78 create new endpoints for managing the tour list

### PR DESCRIPTION
**Pull request by SimerdeepSinghGrewal**
Added Tour.js, tour.routes.js, tour.Controller.js, tour.service.js and tour_helper.js

### CODE REVIEW

File: backend/src/utils/tour_helpers.js [hunk 1]
```diff
@@ -0,0 +1,27 @@
+const validateTriggeringFrequency = (value) => {
```
Comment (erenfn) [inline backend/src/utils/tour_helpers.js:1]: You should use the helper functions in the models 'validate' section. 

For example, replace the validate below with validatePageTargeting. Take a look at #170 if you need a guidance

pageTargeting: {
        type: DataTypes.STRING,
        allowNull: false,
        validate: {
          isIn: [["equals to", "is different from"]],
        },
      },

**Review by erenfn — CHANGES_REQUESTED** [review]

File: backend/src/utils/tour_helpers.js (hunk 1 above)
Comment (SimerdeepSinghGrewal) [reply backend/src/utils/tour_helpers.js:1]: Done but check the error messages if they are right ?

**Review by erenfn — CHANGES_REQUESTED** [review]
Did you test your code, and pull from origin?

node:internal/modules/cjs/loader:1251
bluewave-onboarding-onboarding_backend-1  |   throw err;
bluewave-onboarding-onboarding_backend-1  |   ^
bluewave-onboarding-onboarding_backend-1  |
bluewave-onboarding-onboarding_backend-1  | Error: Cannot find module '../utils/tour_helper'
bluewave-onboarding-onboarding_backend-1  | Require stack:
bluewave-onboarding-onboarding_backend-1  | - /app/src/models/Tour.js
bluewave-onboarding-onboarding_backend-1  | - /app/src/models/index.js
bluewave-onboarding-onboarding_backend-1  | - /app/src/controllers/auth.controller.js
bluewave-onboarding-onboarding_backend-1  | - /app/src/routes/auth.routes.js
bluewave-onboarding-onboarding_backend-1  | - /app/index.js
bluewave-onboarding-onboarding_backend-1  |     at Module._resolveFilename (node:internal/modules/cjs/loader:1248:15)
bluewave-onboarding-onboarding_backend-1  |     at Module._load (node:internal/modules/cjs/loader:1074:27)   
bluewave-onboarding-onboarding_backend-1  |     at TracingChannel.traceSync (node:diagnostics_channel:315:14)
bluewave-onboarding-onboarding_backend-1  |     at wrapModuleLoad (node:internal/modules/cjs/loader:217:24)  
bluewave-onboarding-onboarding_backend-1  |     at Module.require (node:internal/modules/cjs/loader:1339:12) 
bluewave-onboarding-onboarding_backend-1  |     at require (node:internal/modules/helpers:135:16)
bluewave-onboarding-onboarding_backend-1  |     at Object.<anonymous> (/app/src/models/Tour.js:5:5)
bluewave-onboarding-onboarding_backend-1  |     at Module._compile (node:internal/modules/cjs/loader:1546:14)
bluewave-onboarding-onboarding_backend-1  |     at Module._extensions..js (node:internal/modules/cjs/loader:1691:10)
bluewave-onboarding-onboarding_backend-1  |     at Module.load (node:internal/modules/cjs/loader:1317:32) {  
bluewave-onboarding-onboarding_backend-1  |   code: 'MODULE_NOT_FOUND',
bluewave-onboarding-onboarding_backend-1  |   requireStack: [
bluewave-onboarding-onboarding_backend-1  |     '/app/src/models/Tour.js',
bluewave-onboarding-onboarding_backend-1  |     '/app/src/models/index.js',
bluewave-onboarding-onboarding_backend-1  |     '/app/src/controllers/auth.controller.js',
bluewave-onboarding-onboarding_backend-1  |     '/app/src/routes/auth.routes.js',
bluewave-onboarding-onboarding_backend-1  |     '/app/index.js'
bluewave-onboarding-onboarding_backend-1  |   ]
bluewave-onboarding-onboarding_backend-1  | }
bluewave-onboarding-onboarding_backend-1  |
bluewave-onboarding-onboarding_backend-1  | Node.js v22.9.0
bluewave-onboarding-onboarding_backend-1  |
bluewave-onboarding-onboarding_backend-1  | [nodemon] app crashed - waiting for file changes before starting.

File: backend/src/utils/tour_helpers.js (hunk 1 above)
Comment (erenfn) [reply backend/src/utils/tour_helpers.js:1]: They are fine

**Review by erenfn — APPROVED** [review]

**Comment by erenfn** [comment]
Now ready to merge. I fixed the 'tour_helper' import to 'tour_helpers' in Tour.js and changed popup_log to popupLog in index.js (I guess the develop branch wasn't merged correctly).

**Pull request merged**